### PR TITLE
test/e2e: fix checking of nfd-master annotation

### PR DESCRIPTION
**Pull request by marquiz**
The test was broken twofold: Firstly, the annotation was not checked at
all because the name of the node where nfd-master is running was not
set. Secondly, the annotation prefix was used incorrectly.

### CODE REVIEW

**Review by fromanirh — COMMENTED** [review]
LGTM - makes sense to me and matches my admittedly limited experience.
In general, this seems more careful testing so it seems like a good improvement.

File: test/e2e/node_feature_discovery.go [hunk 1]
```diff
@@ -506,6 +506,15 @@ var _ = SIGDescribe("Node Feature Discovery", func() {
 			By("Waiting for the nfd-master pod to be running")
 			Expect(e2epod.WaitTimeoutForPodRunningInNamespace(f.ClientSet, masterPod.Name, masterPod.Namespace, time.Minute)).NotTo(HaveOccurred())
 
+			By("Verifying node where nfd-master is running")
```
Comment (ArangoGutierrez) [inline test/e2e/node_feature_discovery.go:509]: Nit
```suggestion
			By("Verifying the node where nfd-master is running")
```

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**